lib/expression.ex

```diff
@@ -120,7 +120,7 @@ defmodule Expression do
   end
 
   @spec evaluate_as_string!(
-          String.t() | nil,
+          any(),
```

Comment:
This may solve the typespec issue but whatever we're passing in must be string or a nil, otherwise the `parse!` on line 133 will fail.

Reply:
The `parse` that's used in `evaluate_as_string` accepts a string or a number:
https://github.com/turnhub/expression/blob/6e35e4b9e3d0f59d46a1d7fc67195bdba5189667/lib/expression.ex#L86
The `parse_expression` is the one that only accepts a string:
https://github.com/turnhub/expression/blob/6e35e4b9e3d0f59d46a1d7fc67195bdba5189667/lib/expression.ex#L51

The update triggered a dialyzer error in the main repo somewhere that was passing a value for the expression and another method that's consuming that value and expecting it to be `binary() | Number.t()`. I can add a spec identical to the `parse` one with nil like `String.t() | Number.t() | nil`?

Since this is a public repo, not sure if I should paste links to where it's failing. I'll add it on Slack and tag you.

Reply:
I replaced `@any` with `String.t() | Number.t() | nil` to match the parser and also allow nil which we recently supported.

Reply:
@smn I have changed `Expression.parse!/1` to accept numbers in #129. So `Expression.evaluate_as_string!/1` should also accept numbers. In that case, the spec should be `String.t() | Number.t() | nil` instead of `any()` as @maryamaljanabi did in [0f05eaa](https://github.com/turnhub/expression/pull/238/commits/0f05eaa560143cd7ca7370bf090eee3806fd35f8).